web/src/filter.ts

```diff
@@ -481,6 +481,40 @@ export class Filter {
         return terms;
     }
 
+    static is_valid_search_pill(term: NarrowTerm): boolean {
+        switch (term.operator) {
+            case "has":
+                return ["image", "link", "attachment", "reaction"].includes(term.operand);
+            case "is":
+                return [
+                    "dm",
+                    "private",
+                    "starred",
+                    "mentioned",
+                    "alerted",
+                    "unread",
+                    "resolved",
+                    "followed",
+                ].includes(term.operand);
+            case "in":
+                return ["home", "all"].includes(term.operand);
+            case "id":
+            case "near":
+                return Number.isInteger(Number(term.operand));
+            case "channel":
+            case "stream":
+                return stream_data.get_sub(term.operand) !== undefined;
+            case "sender":
+            case "dm":
+            case "dm-including":
+                return term.operand
+                    .split(",")
+                    .every((email) => people.get_by_email(email) !== undefined);
+            default:
+                return true;
```

Comment:
It might be better to be more explicit about all the valid operators and return false for `default`